DependencyInjection/Compiler/ProviderCompilerPass.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Swarrot\SwarrotBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class ProviderCompilerPass implements CompilerPassInterface
+{
+    /** {@inheritDoc} */
+    public function process(ContainerBuilder $container)
+    {
+        if ($container->has('swarrot.factory.default') || !$container->hasParameter('swarrot.provider_config')) {
+            return;
+        }
+
+        $providersIds = [];
+
+        foreach ($container->findTaggedServiceIds('swarrot.provider_factory') as $id => $tags) {
+            foreach ($tags as $tag) {
+                $providersIds[isset($tag['alias']) ? $tag['alias'] : $id] = $id;
```

Comment:
@jderusse and for the FormPass, this won't be supported anymore in 3.0. Using the service id as fallback when the alias is not specified was a mistake here (and we need to keep it in 2.0 for BC reasons).

Having the user configure it as ``pecl`` is better than forcing it to know the internal id of the service (and it is needed for BC reasons btw).